src/libaktualizr/primary/sotauptaneclient.cc

```diff
@@ -940,7 +940,10 @@ result::Install SotaUptaneClient::uptaneInstall(const std::vector<Uptane::Target
     std::vector<Uptane::Target> primary_updates = findForEcu(updates, primary_ecu_serial);
 
     //   6 - send metadata to all the ECUs
-    sendMetadataToEcus(updates);
+    if (!sendMetadataToEcus(updates)) {
+      result.dev_report = {false, data::ResultCode::Numeric::kInternalError, "Metadata verification failed"};
+      return std::make_tuple(result, "Secondary metadata verification failed", false);
+    }
```

Comment:
Just confirming: if this fails, does the backend recognize it and abort the update, or does it stay pending and cause aktualizr to retry it?

Reply:
It's actually a good question. It's out of the scope of the given PR nevertheless I think it makes sense to return something more meaningful to `putMetadata`, not just bool like we do for `install`. So, the follow-up behavior can depend on an error type, e.g. if it's some network-related error then it makes sense to retry if Uptane verification of metadata actually fails then there is no point to retry.

I suppose proper testing of the actualize behavior after putMetadata failure requires adding test(s) to the python-based tests and OTF tests in order to see how it behaves if the uptane cycle is running.

Reply:
@patrickvacek I just had a tried and the installation shows as failed on the UI, with nothing pending.

However, if I re-run `aktualizr check` after the failure, it still shows 1 new update to install, which it will do if `aktualizr once` is called without the injected error.

So it's not ideal but it sounds like we can't really fix that on the client?

Reply:
I think this actually might require a client-side fix. I'm wondering if we need to drop the Director Targets metadata in this case (and all the similar returns for the metadata checks and such above this in this function). I want to avoid more of these cases where we get stuck in a loop, and as long as the empty targets optimization is in place, we have to work around it.

Reply:
Ah you're right, it doesn't persist if we drop the metadata. That's what I pushed in the last version.

Reply:
Glad to know that fixed it, but annoying that we have to keep expanding this target dropping business. I suspect all of the errors in that lambda function should return true for `drop_targets`.